index.js

```diff
@@ -5,41 +5,108 @@
 
 // This script is modified version of
 // https://github.com/brave/brave-ios/blob/development/Client/Frontend/UserContent/UserScripts/Playlist.js
-(function () {
+medias = (async function () {
   // This will be replaced by native code on demand.
   const siteSpecificDetector = null
 
+  /**
+   * Returns a Promise that resolves with a boolean argument indicating whether
+   * the `src` URL could be a blob pointing at a MediaSource object.
+   * @param {string} src
+   * @returns {Promise}
+   */
+  async function isMediaSourceObjectURL (src) {
+    if (!src || !src.startsWith('blob:')) {
+      return false
+    }
+
+    const controller = new AbortController()
+    const signal = controller.signal
+    let timeout
+    const maybeAbortFetch = new Promise(resolve =>
+      timeout = setTimeout(() => {
+        resolve(false)
+        controller.abort()
+      }, 500)
+    )
+
+    return Promise.any([
+      new Promise(resolve => {
+        fetch(src, { signal })
+          .then(() => false)
+          .catch(() => true)
```

Comment:
Unfortunately, this didn't resolve()

Reply:
Ah, that's because it was still wrapped in a `new Promise(r => {})` - I think we just need

```tsx
await Promise.any([fetch().then(() => false).catch(() => true), abortPromise])

//instead of 
await Promise.any([new Promise(() => ...), abortPromise])
```